src/esp/io/URDFParser.h

```diff
@@ -0,0 +1,596 @@
+  std::string m_materialName;
+};
+
+//! Defines some collision shape flags
+enum CollisionFlags {
+  //! The shape has a custom group defined (e.g. <collision group="32">)
+  HAS_COLLISION_GROUP = 1,
+  //! The shape has a custom mask defined (e.g. <collision mask="32">)
+  HAS_COLLISION_MASK = 2,
+};
+
+//! Parsed from a URDF link collision shape (<collision>)
+struct CollisionShape : Shape {
+  int m_flags{0};
+  //! custom collision group (e.g. <collision>)
+  int m_collisionGroup;
+  //! custom collision mask
+  int m_collisionMask;
+  CollisionShape() = default;
+};
+
+//! Parsed from a URDF link inertia properties (<inertial>). Stores dynamic
+//! properties of ths link.
+struct Inertia {
+  //! local transform of the link in parent joint space
+  Magnum::Matrix4 m_linkLocalFrame{0.0};
+  bool m_hasLinkLocalFrame;
+
+  //! mass of the link (0 mass indicates unmovable, static object)
+  double m_mass{0.0};
+
+  //! inertia matrix upper triangular entries. Computed automatically if not
+  //! specified.
+  double m_ixx, m_ixy, m_ixz, m_iyy, m_iyz, m_izz = 0.0;
+
+  Inertia() = default;
+};
+
+//! Parsed from a URDF joint (<joint>), connects two links.
+struct Joint {
+  //! Joint name
+  std::string m_name;
+  //! Type of the joint, used to pivot on relevant fields for instancing
+  JointTypes m_type;
+  //! Relative transform of the joint from its parent link
+  Magnum::Matrix4 m_parentLinkToJointTransform;
+  //! Name of the parent link and key in the link cache
+  std::string m_parentLinkName;
+  //! Name of the child link and key in the link cache
+  std::string m_childLinkName;
+  //! Some joints (e.g. revolute) have a local axis (e.g. X, Y, Z)
+  Magnum::Vector3 m_localJointAxis;
+
+  //! Joint limits if provided
+  double m_lowerLimit{0};
+  double m_upperLimit{-1};
+
+  //! Joint force/torque limits if provided
+  double m_effortLimit{0};
+  double m_velocityLimit{0};
+
+  //! Joint damping and friction. Used to generate default JointMotors for
+  //! BulletArticulatedObjects.
+  double m_jointDamping{0};
+  double m_jointFriction{0};
+  Joint() = default;
+};
+
+//! Contact info cached for links, but not currently used
+enum LinkContactFlags {
+  CONTACT_HAS_LATERAL_FRICTION = 1,
+  CONTACT_HAS_INERTIA_SCALING = 2,
+  CONTACT_HAS_CONTACT_CFM = 4,
+  CONTACT_HAS_CONTACT_ERP = 8,
+  CONTACT_HAS_STIFFNESS_DAMPING = 16,
+  CONTACT_HAS_ROLLING_FRICTION = 32,
+  CONTACT_HAS_SPINNING_FRICTION = 64,
+  CONTACT_HAS_RESTITUTION = 128,
+  CONTACT_HAS_FRICTION_ANCHOR = 256,
+};
+
+//! Contact info cached for links, but not currently used (parsed from
+//! <contact>)
+struct LinkContactInfo {
+  float m_lateralFriction{0.5};
+  float m_rollingFriction{0};
+  float m_spinningFriction{0};
+  float m_restitution{0};
+  float m_inertiaScaling{1};
+  float m_contactCfm{0};
+  float m_contactErp{0};
+  float m_contactStiffness{1e4};
+  float m_contactDamping{1};
+
+  int m_flags;
+
+  LinkContactInfo() { m_flags = CONTACT_HAS_LATERAL_FRICTION; }
```

Comment:
Why not use just make this the default initialization? Then the struct would be trivially constructible?